### PR TITLE
Automate Release Process

### PR DESCRIPTION
**Pull request by ColinKYuen**
### Summary

Automate Release Process

### Description

Automates the release process using GitHub Actions & change to `CHANGES` to follow style of https://keepachangelog.com/en/1.0.0/

When a tag is created, this workflow will run
- Build & Test Drivers 
- Sign Jars with GPG
- Parse the top version of CHANGES for body of release
- Publish & Release

### Sample Release
Screen Shot 2021-11-10 at 2 41 28 PM

### Additional Reviewers
@karenc-bq 
@matthewh-BQ 
@sergiyv-bitquill

### CODE REVIEW

**Comment by ColinKYuen** [comment]
Example release that was generated.
https://github.com/Bit-Quill/aws-mysql-jdbc/releases/tag/untagged-88cdd6fdd628c5853e23

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          export GPG_TTY=$(tty)
+          for jar in build/libs/*.jar; do gpg --detach-sign --armor $jar; done
+      - name: 'Upload to Release'
+        uses: ncipollo/release-action@v1
+        with:
+          draft: true
+          name: "AWS JDBC Driver for MySQL Public Preview - v${{ env.RELEASE_VERSION }}"
+          bodyFile: RELEASE_DETAILS.md
+          artifacts: build/libs/*
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (karenc-bq) [inline .github/workflows/release.yml:69]:
```suggestion
          token: ${{ secrets.GITHUB_TOKEN }}
          
```

File: CHANGES [hunk 2]
```diff
@@ -1,18 +1,13 @@
...
-
   - Added clarifications and improvements to README.md.
 
-Version 0.1.0 (Public Preview)
-
+## Version 0.1.0 (Public Preview) - 2021-01-06
+### Added
   - Based on the MySQL 8.0.21 community driver.
-
   - The driver is cluster aware for Amazon Aurora MySQL. It takes advantage of Amazon Aurora’s fast failover capabilities, reducing failover times from minutes to seconds.
```
Comment (karenc-bq) [inline CHANGES:13]: This change log doesn't really make sense. @matthewh-BQ Any suggestions?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          export RELEASE_DETAILS="$(sed '$d' <<< "$RELEASE_DETAILS")"
+          touch RELEASE_DETAILS.md
+          echo "$RELEASE_DETAILS" > RELEASE_DETAILS.md
+      - name: 'Install GPG Secret Key'
+        id: install-secret-key
+        run: |
+          cat <(echo -e "${{ secrets.GPG_SECRET_KEY }}") | gpg --batch --import
+      - name: 'Sign Jars'
+        run: |
+          export GPG_TTY=$(tty)
```
Comment (karenc-bq) [inline .github/workflows/release.yml:47]: Do we still need this if the GPG key doesn't require a password?

File: CHANGES [hunk 4]
```diff
@@ -1,18 +1,13 @@
...
-
   - Added "setAwsProtocolOnly" static method.
-
   - Merged upstream changes from MySQL 8.0.23 community driver.
-
   - Added clarifications and improvements to README.md.
 
-Version 0.1.0 (Public Preview)
-
+## Version 0.1.0 (Public Preview) - 2021-01-06
```
Comment (karenc-bq) [inline CHANGES:10]: https://github.com/awslabs/amazon-qldb-driver-java/blob/master/CHANGELOG.md
Example from QLDB, we can follow their example for version 2.0.0.
Don't forget to link the tags.

File: .github/workflows/release.yml (hunk 3 above)
Comment (ColinKYuen) [reply .github/workflows/release.yml:47]: Yes, it's used for signing.

**Review by sergiyv-bitquill — APPROVED** [review]

**Review by matthewh-BQ — APPROVED** [review]

**Pull request merged**